### PR TITLE
Upload files to bucket from Image Pipeline, enhance logging (SCP-4667)

### PR DESCRIPTION
**Pull request by eweitz**
This lets us upload files from a PAPI VM to Google Cloud Storage buckets.  It works for both images and log files.

**Images**
WEBP image files of gene expression scatter plots are also now uploaded.  The current implementation is crude but effective.  I propose optimizations that I've ticketed as a separate task.

**Log file**
Being able to upload (a.k.a. delocalize) a log file from PAPI gives us observability for completed Image Pipeline jobs.  Log files are more detailed than the job run metadata available via `gcloud alpha genomics operations describe ${OPERATION_ID}`.  This is needed to debug errors that occur outside local machines, e.g. `403 Forbidden` responses due to authorization issues accessing the staging SCP REST API from the staging GCP project (covered in separate in-progress work).

[Previously](https://github.com/broadinstitute/single_cell_portal_core/pull/1619), logs just went to the `stdout` via `console.log`.  Now, they also write to the `log.txt` file with timestamps.  The Node command that was run is also logged.

Importantly, these logs are _streamed_ to the bucket.  This gives observability into _ongoing_ jobs, and more crash-resiliently.

**To test**

* Execute steps 1-4 from test instructions in #1619
* Run Image Pipeline via PAPI with current development image:
> `gcloud alpha genomics pipelines run --regions us-east1 --command-line 'export NODE_TLS_REJECT_UNAUTHORIZED=0; node expression-scatter-plots.js --accession="SCP303" --environment="staging" --cores="1" --debug-headless; # human milk DE pilot' --docker-image gcr.io/broad-singlecellportal-staging/image-pipeline:0.0.1_047029631 --service-account-email $SA_EMAIL_ADDRESS # Command to run Image Pipeline on PAPI`
* Run steps 7-8 from PR linked above
* Note failure status code, which is expected
* Go to the [expected bucket file path](https://console.cloud.google.com/storage/browser/_details/broad-singlecellportal-staging-testing-data/_scp_internal/parse_logs/log_image_pipeline.txt)
* Confirm timestamp on page approximately matches time you ran this test
* Click file to view it in web browser
* Confirm log file has timestamps, the Node command you ran, and a stack trace of the error, e.g.:
```
[2022-09-26T16:58:42.858Z] -- : Command run via Node:

expression-scatter-plots.js --accession=SCP303 --environment=staging --cores=1 --debug-headless

[2022-09-26T16:58:42.860Z] -- : Number of CPUs to be used on this client: 1
[2022-09-26T16:58:42.863Z] -- : Accession: SCP303
[2022-09-26T16:58:42.864Z] -- : Fetching https://singlecell-staging.broadinstitute.org/single_cell/api/v1/studies/SCP303/explore?isImagePipeline=true
[2022-09-26T16:58:43.117Z] -- : response.status
[2022-09-26T16:58:43.118Z] -- : 403
[2022-09-26T16:58:43.124Z] -- : response text
[2022-09-26T16:58:43.128Z] -- : <!doctype html><meta charset="utf-8"><meta name=viewport content="width=device-width, initial-scale=1"><title>403</title>403 Forbidden
[2022-09-26T16:58:43.129Z] -- : SyntaxError: Unexpected token < in JSON at position 0
    at JSON.parse (<anonymous>)
    at run (file:///image-pipeline/expression-scatter-plots.js:414:17)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async file:///image-pipeline/expression-scatter-plots.js:476:3
```

(The error accessing staging is a separate issue -- a big part of this ticket is getting observability into errors that happen in the PAPI VM, as demonstrated above.)

This satisfies SCP-4667.

### CODE REVIEW

**Comment by bistline** [comment]
Code looks fine, but I'm unable to progress because I get this error:
```
manifest for gcr.io/broad-singlecellportal-staging/image-pipeline:0.0.1_04702963 not found: manifest unknown: Failed to fetch \"0.0.1_04702963\" from request \"/v2/broad-singlecellportal-staging/image-pipeline/manifests/0.0.1_04702963\"
```

I can see that the [image](https://console.cloud.google.com/gcr/images/broad-singlecellportal-staging/global/image-pipeline@sha256:f1777c4f57c476cf6e60a97117483c937c084ab8ab2941ce2f2b07830754e7e1/details?project=broad-singlecellportal-staging) is in GCR, but for some reason I'm unable to pull it for this job.

**Review by bistline — APPROVED** [review]

File: image-pipeline/expression-scatter-plots.js [hunk 1]
```diff
@@ -403,13 +429,58 @@ let startTime
...
+  print(
+    `File "${fromFilePath}" uploaded to destination "${toFilePath}" ` +
+    `in bucket "${bucketName}"`,
+    preamble
+  )
+}
+
+/** Upload / delocalize log file to GCS bucket */
+async function uploadLog() {
+  await uploadToBucket('log.txt', 'parse_logs/log_image_pipeline.txt')
```
[the remark this answers is not in the document]
Comment (eweitz) [reply image-pipeline/expression-scatter-plots.js:458]: I'm unsure, too.  How about a miillisecond-granularity timestamp, e.g. 202209281009123?

[the remark this answers is not in the document]
Comment (eweitz) [reply image-pipeline/expression-scatter-plots.js:458]: I'm unsure, too. How about a millisecond-granularity timestamp, e.g. 202209281009123?

**Review by jlchang — APPROVED** [review]
Functional review: Works as described (It took me a second to realize the timestamps in the logs were UTC! 

Streaming the log file to the bucket is a great feature to improve robustness of logging and facilitate troubleshooting - it will be great to replicate this functionality in Ingest Pipeline.

**Pull request merged**